google/cloud/forseti/common/gcp_api/cloudasset.py

```diff
@@ -165,10 +217,25 @@ def export_assets(self, parent, destination_object, content_type=None,
             raise ValueError('parent must start with folders/, projects/, or '
                              'organizations/')
 
+        if not destination_object and not output_config:
+            raise ValueError('Either destination_object or output_config must '
```

Comment:
qq: If we are raising these `ValueError`, are they being handled correctly?

For example:
#1: is the log message appropriate when it references `root_resource_id`?
https://github.com/forseti-security/forseti-security/blob/1e60285d3fbb8b1ae1b2b8333496d24a31ac520e/google/cloud/forseti/services/inventory/base/cloudasset.py#L199-L201

#2: not seeing `ValueError` being handled
https://github.com/forseti-security/forseti-security/blob/a9a820feb400dca968eef941fc4d0fc1d8f97ba7/google/cloud/forseti/common/gcp_api/cloudasset.py#L176-L182

Reply:
If a value error is raised, it is due to an unexpected bad usage of the function and the resulting code should blow up. This should be always be caught in tests. If it happens live then something is fundamentally wrong with the way the method is called, and that should also result in an exception bubbling up and killing the process.

Reply:
In other words, in cloudasset.py the ValueError may be better re-raised and cause the process to die rather then to silently handle the exception and log, because in that case the server is misconfigured and falling back to direct API querying is likely not what the user intended.

I'm going to be refactoring the cloudasset.py code in a future PR and will address this then.